runners/google-cloud-dataflow-java/worker/build.gradle

```diff
@@ -54,6 +54,7 @@ def sdk_provided_project_dependencies = [
         ":runners:google-cloud-dataflow-java",
         ":sdks:java:extensions:avro",
         ":sdks:java:extensions:google-cloud-platform-core",
+        ":sdks:java:io:kafka", // For metric propagation into worker
```

Comment:
This is bad code coupling. Fetching static name isn't sufficient justification to introduce this mandatory dependency to Dataflow worker jar. This also makes confluent repository mandatory for user project.